cmd/add.go

```diff
@@ -100,5 +121,8 @@ func cmdAddAccount(ctx context.Context, sherlock *internal.Sherlock) *cobra.Comm
 	addGroup.Flags().StringVarP(&opts.tag, "tag", "t", "", "optional tag for this account")
 	addGroup.Flags().BoolVarP(&opts.insecure, "insecure", "i", false, "allow insecure group password")
 
+	// I set this to string to make input validation checking easier if the input data is not a valid number
+	addGroup.Flags().StringVarP(&opts.gen, "gen", "e", "", "length for auto-generate secure passsword. Create your own password when not set")
```

Comment:
Yes I thought the same way as you at first time. But I then realized that it makes a new problem. 

So, if you use `Flags().IntVarP()` for the --gen or -e when creating account, and you pass the argument a number, then it will work just fine. But, if you pass the argument a string, then things start to a little bit messy. the error thrown will be cobra's error because it won't receive non numeric input. So why I use string to get the argument because with it, I can pass proper error message to the user if they accidently pass non numeric as a password length. maybe you can try with yourself. try to change line 125 with `Flags().IntVarP()` and then try to pass non numeric argument.

But if you insist to use `Flags().IntVarP()`, that's a no problem for me, I will make the change as you request. 

CMIIW.

thank you.

Reply:
Thanks for pointing this out - was not aware `cobra` has this weird error message 😄
LGTM 🎉